### PR TITLE
Issue 60 create branch api tests

### LINKED ISSUE #60: Create branch api tests
## Basic Information
Create tests for all the branch endpoints (I think there's only one). Make sure to also test negative test cases. Tests already exist for testing the soft delete, so they go in that file

## Other Information
**[Are there any additional information that we should know about?]**

### PR DESCRIPTION
**Pull request by Divrr**
## Change Summary
Created tests for expected behaviour of the branch API endpoints, as well as "negative" tests for missing field, invalid field, unauthorised, and does-not-exist errors.

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [ ] The change has documentation

## Other Information
`"/workspaces/coexist/server/api/branch/tests.py", line 145` fails, which is testing a PATCH of an invalid field. I can't tell if this is intended functionality?

# Related issue

- Resolve #60

### CODE REVIEW

**Comment by Divrr** [comment]
PLEASE NOTE: THE BACKEND CODE CHECK FAILING IS SOMETHING THAT REQUIRES REVIEW:

> "/workspaces/coexist/server/api/branch/tests.py", line 145 fails, which is testing a PATCH of an invalid field. I can't tell if this is intended functionality?

File: server/api/branch/tests.py [hunk 1]
```diff
@@ -33,3 +34,132 @@ def test_GetAllExceptis_deleted(self):
...
+        response = self.client.patch(f'/api/branch/{self.branch.id}/', {
+            'name': ''
+        })
+        self.assertEqual(response.status_code, 400)
+
+        # "invalid field" case
+        response = self.client.patch(f'/api/branch/{self.branch.id}/', {
+            'invalid': 'invalid'
+        })
+        self.assertEqual(response.status_code, 400)
```
Comment (dct0) [inline server/api/branch/tests.py:145]: It seems that the default behaviour for the UpdateModelMixin is to ignore extra fields when PATCHing. Instead, you could test an "invalid field" by attempting to update a field with the wrong type. Otherwise it's fine to 200

**Review by dct0 — CHANGES_REQUESTED** [review]
Nice work. Just a comment about that pathc test case

File: server/api/branch/tests.py (hunk 1 above)
Comment (Divrr) [reply server/api/branch/tests.py:145]: Ah ok, DRF decided that's not a biggie, ig. Thanks for looking that up!

**Review by dct0 — APPROVED** [review]
